plugins/modules/cloud/misc/terraform.py

```diff
@@ -360,12 +385,25 @@ def build_plan(command, project_path, variables_args, state_file, targets, state
         return plan_path, False, out, err, plan_command if state == 'planned' else command
     elif rc == 1:
         # failure to plan
-        module.fail_json(msg='Terraform plan could not be created\r\nSTDOUT: {0}\r\n\r\nSTDERR: {1}'.format(out, err))
+        module.fail_json(
+            msg='Terraform plan could not be created\r\nSTDOUT: {out}\r\n\r\nSTDERR: {err}\r\nCOMMAND: {cmd} {args}'.format(
+                out=out,
+                err=err,
+                cmd=' '.join(plan_command),
+                args=' '.join([shlex_quote(arg) for arg in variables_args])
+            )
+        )
     elif rc == 2:
         # changes, but successful
         return plan_path, True, out, err, plan_command if state == 'planned' else command
 
-    module.fail_json(msg='Terraform plan failed with unexpected exit code {0}. \r\nSTDOUT: {1}\r\n\r\nSTDERR: {2}'.format(rc, out, err))
+    module.fail_json(msg='Terraform plan failed with unexpected exit code {rc}. \r\nSTDOUT: {out}\r\nSTDERR: {err}\r\nCOMMAND: {cmd} {args}'.format(
```

Comment:
The `\r` should not be necessary, in fact now that I see it I reckon that's probably some cases to fix. This code will run on systems other than Windows, and in code, we should alwyas use `\n` only.

Reply:
Fixed these.